windows-agent/internal/daemon/networking_windows.go

```diff
@@ -35,6 +65,18 @@ func getWslIP() (net.IP, error) {
 	return nil, fmt.Errorf("could not find WSL adapter")
 }
 
+// networkIsMirrored detects whether the WSL network is mirrored or not.
+func networkIsMirrored() (bool, error) {
+	// It does so by launching the system distribution.
+	cmd := exec.CommandContext(context.Background(), "wsl", "--system", "wslinfo", "--networking-mode", "-n")
+
+	out, err := cmd.Output()
+	if err != nil {
+		return false, fmt.Errorf("could not get networking mode: %w", err)
```

Comment:
The error here will not tell you much (only which error code the program exited with). You probably want to print stdout + stderr.